### PR TITLE
Add ServiceProvider extensions

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Add ServiceProvider extensions from the designer code. For try methods allow for null provider. Add internal overloads to Control.

Annotate PropertyTab and use the new overloads in some of the PropertyGrid classes.

Handle unexpected/null ISite.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5247)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
x86 debug failed with #4813

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.Annotated.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        /// <summary>
+        ///  Gets the service object of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The type of the service object to get.</typeparam>
+        /// <param name="service">If found, contains the service object when this method returns; otherwise, <see langword="null"/>.</param>
+        /// <returns>A service object of type <typeparamref name="T"/> or <see langword="null"/> if there is no such service.</returns>
+        internal bool TryGetService<T>(
+            [NotNullWhen(true)] out T? service)
+            where T : class
+            => (service = GetService(typeof(T)) as T) != null;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.Annotated.cs:29]: ditto

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs [hunk 2]
```diff
@@ -3289,39 +3264,35 @@ private void OnViewButtonClickPP(object sender, EventArgs e)
...
-                                    }
-                                    finally
-                                    {
-                                        SetFlag(Flags.InternalChange, false);
-                                    }
+                                try
+                                {
+                                    // Now notify the change service that the change was successful.
+                                    SetFlag(Flags.InternalChange, true);
+                                    changeService.OnComponentChanged(baseObject, null, null, null);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:3291]: FYI: we could add an extension method that takes only component as a parameter and uses `null`s aa default values for other parameters.

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:3291]: I'll take a look in the next steps.

**Pull request merged**